pkg/controller/labelidentity/label_group_index.go

```diff
@@ -299,45 +313,26 @@ func (i *LabelIdentityIndex) deleteSelector(selectorKey string, policyKey string
 	}
 }
 
-// SetPolicySelectors registers ClusterSet-scope policy selectors with the labelIdentityIndex,
-// and then retrieves all the LabelIdentity IDs that currently match these selectors.
-func (i *LabelIdentityIndex) SetPolicySelectors(selectors []*types.GroupSelector, policyKey string) []uint32 {
-	var labelIdentityIDs []uint32
+// SetPolicySelectors registers a policy's latest selectors with the index.
+// For the selectors that the policy no longer possess, it removes the policy key
+// from those selectorItems.
+func (i *LabelIdentityIndex) SetPolicySelectors(selectors []*types.GroupSelector, policyKey string) {
 	newSelectors := map[string]*types.GroupSelector{}
 	for _, s := range selectors {
-		klog.V(4).InfoS("Getting matched LabelIdentity for policy selector", "selector", s.NormalizedName, "policy", policyKey)
 		newSelectors[s.NormalizedName] = s
 	}
 	originalSelectors := i.getPolicySelectors(policyKey)
```

Comment:
Considering most policies do not have clusterSet scope selectors, it may be more efficient to have `getPolicySelectors()` return nil in such cases. The same for `newSelectors`.

Reply:
Done. Also added early return for the function when `originalSelectors` is empty